### PR TITLE
initial rewrite for one-week dismissals

### PR DESCRIPTION
**Pull request by Pomax**
Closes https://github.com/mozilla/foundation.mozilla.org/issues/3324

This switches the donate modal to one that can be dismissed for a week at a time. It will pop up after 10 seconds, and will keep popping up until dismissed. Once dismissed, the time of dismissal is stored in localStorage, and once a week has passed, the modal will start popping up again until dismissed again.

### CODE REVIEW

**Comment by kristinashu** [comment]
Looks good at all screen widths except at super narrow (<335px) when the button gets cut off. But I think that's ok because I doubt anyone would go that narrow.

image

File: source/js/donate-modal/donate-modal.jsx [hunk 1]
```diff
@@ -2,58 +2,47 @@ import React from "react";
...
+    if (lastLoad) {
+      var diff = (Date.now() - parseInt(lastLoad, 10)) / DAY;
+      if (diff < DISMISSAL_DELAY) {
+        dismissed = true;
+      }
+    }
 
     this.state = {
-      delay: DELAY - prevTimer,
+      delay: INITIAL_DELAY,
```
Comment (mmmavis) [inline source/js/donate-modal/donate-modal.jsx]: if value of `delay` won't be changed, should we turn it into a class property instead a state?

File: source/js/donate-modal/donate-modal.jsx [hunk 2]
```diff
@@ -2,58 +2,47 @@ import React from "react";
...
+const localStorage = typeof window === "undefined" ? {} : window.localStorage;
 
 class DonateModal extends React.Component {
   constructor(props) {
     super(props);
 
-    let dismissed = !!sessionStorage.getItem(KEY_STATE) || false;
-    let prevTimer = this.getPrevTimer();
+    const INITIAL_DELAY = parseInt(this.props.delay) || 10 * MILLISECONDS; // 10 seconds, in ms
+    const DISMISSAL_DELAY = parseInt(this.props.hideFor) || 7; // one week, in days
```
Comment (mmmavis) [inline source/js/donate-modal/donate-modal.jsx]: I was thinking maybe we could rename these two variables into `INITIAL_DELAY_IN_MILLISECONDS` and `DISMISSAL_DELAY_IN_DAYS` for clarity?

File: source/js/donate-modal/donate-modal.jsx (hunk 1 above)
Comment (Pomax) [reply source/js/donate-modal/donate-modal.jsx]: it can be changed through a prop, though.

File: source/js/donate-modal/donate-modal.jsx (hunk 2 above)
Comment (Pomax) [reply source/js/donate-modal/donate-modal.jsx]: good call

**Pull request merged**